### PR TITLE
fix TCP transport for web

### PR DESCRIPTION
**Pull request by lukediamond**
Makes sure the correct transport is chosen for web and node, adds shims for `net`/`tls`.

@misterupkeep

### CODE REVIEW

File: scripts/build.ts
Comment (misterupkeep) [inline scripts/build.ts]: Why, though? The rest of the script file relies on gnutils anyways, might as well write it as `mkdir -p dist/npm`

File: src/Client.ts
Comment (misterupkeep) [inline src/Client.ts]: TCP should also be enabled by default
```suggestion
                tcp: true,
```

**Review by misterupkeep — CHANGES_REQUESTED** [review]
TCP needs to be enabled by default, and I have to test this in both browser and node environments before merging.

File: scripts/build.ts
Comment (lukediamond) [reply scripts/build.ts]: Because I'm running windows right now and I would rather replace this entire file with `fs` calls than get that set up and make it a dependency. It's completely unnecessary to call the shell here.

**Comment by lukediamond** [comment]
Can we merge this?

**Comment by misterupkeep** [comment]
I still have to test the actual TCP transport. I know it works (or rather, does _not_ work) in the browser, but haven't so much as tried connecting over TCP. 

Perhaps hooking up TCP into Electron builds would be the best, most adequate test, but _that_ is waiting on bonfire-xmpp/bonfire#60, which is in turn waiting on upstream.

Merging without testing (or taking your word) might be pushing unstable code to what should be a stable library. We can merge but wait on the release until we get TCP working on Bonfire. Even if it's broken, it should be working sans TCP transport.

**Pull request closed without merging**
